graphix/pauli.py

```diff
@@ -227,7 +258,20 @@ def matrix(self) -> np.ndarray:
         """
         return self.__unit.complex * graphix.clifford.CLIFFORD[self.__symbol.value + 1]
 
-    def __repr__(self):
+    def get_eigenstate(self, eigenvalue=0) -> BasicStates:
+        from graphix.states import BasicStates
+
+        if self.symbol == IXYZ.X:
+            return BasicStates.PLUS if eigenvalue == 0 else BasicStates.MINUS
+        elif self.symbol == IXYZ.Y:
+            return BasicStates.PLUS_I if eigenvalue == 0 else BasicStates.MINUS_I
+        elif self.symbol == IXYZ.Z:
+            return BasicStates.ZERO if eigenvalue == 0 else BasicStates.ONE
+        # Any state is eigenstate of the identity
+        elif self.symbol == IXYZ.I:
+            return BasicStates.PLUS
+
+    def __repr__(self) -> str:
```

Comment:
It should be the case: return values will be of the form `I`, `X`, `Y` or `Z` (declared at top-level in the module), prefixed by `-` and/or by `1j * `.

Reply:
It is not enough as `repr(...)` is evaluated at the caller's scope.
How about prefixing it with the class name?

Reply:
Why not use either...

- `__str__` : it can be freely designed
- use a standard design for `__repr__` : something like `f"{type(self).__name__}({...})"`

Reply:
Fixed in 8a1c34d, where I added the prefix `graphix.pauli.`. It’s important to note that the caller's scope is not pertinent here. As you mentioned earlier, it would be bad practice for the caller to execute the result of `repr(...)`, and we cannot make any assumptions about the caller's environment anyway. The Python documentation mentions [_given an appropriate environment_](https://docs.python.org/3/reference/datamodel.html#object.__repr__), meaning that the goal is to easily copy and paste the printed value somewhere else (where we don’t know anything about the scope, just reasonable assumptions).

Overriding `__repr__` instead of defining a separate `as_str` method is very convenient, especially for debugging purposes. If we don’t override `__repr__`, the Pauli object would be printed as `<graphix.pauli.Pauli object at 0x...>`, which is neither evaluable nor readable.